src/phases/weather-effect-phase.ts

```diff
@@ -38,60 +42,88 @@ export class WeatherEffectPhase extends CommonAnimPhase {
+   * @returns `true` if the weather is suppressed
+   *
+   * @todo Merge this logic with {@linkcode Weather.isEffectSuppressed}. This method applies the suppressing attributes directly
+   * while `isEffectSuppressed` uses a `hasAbility` check to avoid flyouts appearing at the wrong time. These may
+   * be combined by rewriting `isEffectSuppressed` to support simulated application.
+   */
+  private tryCancelWeatherEffects(weather: Weather): boolean {
     const cancelled = new BooleanHolder(false);
 
     this.executeForAll((pokemon: Pokemon) =>
       applyAbAttrs<SuppressWeatherEffectAbAttr>(AbAttrFlag.SUPPRESS_WEATHER_EFFECT, pokemon, false, weather, cancelled),
     );
 
-    if (cancelled.value) {
-      return end();
-    }
-
-    const inflictDamage = (pokemon: Pokemon): void => {
-      const cancelled = new BooleanHolder(false);
+    return cancelled.value;
+  }
 
-      applyAbAttrs<PreWeatherDamageAbAttr>(AbAttrFlag.PRE_WEATHER_DAMAGE, pokemon, false, weather, cancelled);
-      applyAbAttrs<BlockNonDirectDamageAbAttr>(AbAttrFlag.BLOCK_NON_DIRECT_DAMAGE, pokemon, false, cancelled);
+  /**
+   * Inflicts damage on the given Pokemon from the end-of-turn effects
+   * of specific {@linkcode WeatherType | types of weather}. This accounts for
+   * immunity to weather-based damage from the Pokemon's typing, abilities, and
+   * other effects.
+   * @param weather - The current {@linkcode Weather} on the field
+   * @param pokemon - The {@linkcode Pokemon} to which damage may apply
+   */
+  private tryInflictWeatherDamage(weather: Weather, pokemon: Pokemon): void {
+    if (!weather.isDamaging()) {
+      return;
+    }
 
-      if (cancelled.value || pokemon.getTag(BattlerTagType.UNDERGROUND) || pokemon.getTag(BattlerTagType.UNDERWATER)) {
-        return;
-      }
+    const cancelled = new BooleanHolder(false);
 
-      const damage = toDmgValue(pokemon.getMaxHp() / 16);
+    applyAbAttrs<PreWeatherDamageAbAttr>(AbAttrFlag.PRE_WEATHER_DAMAGE, pokemon, false, weather, cancelled);
+    applyAbAttrs<BlockNonDirectDamageAbAttr>(AbAttrFlag.BLOCK_NON_DIRECT_DAMAGE, pokemon, false, cancelled);
 
-      globalScene.phaseManager.queueMessagePhase(getWeatherDamageMessage(weather.weatherType, pokemon) ?? "");
-      pokemon.damageAndUpdate(damage, { result: HitResult.EFFECTIVE, preventEndure: true });
-    };
+    if (
+      cancelled.value
+      || pokemon.getTypes(true, true).some((t) => weather.isTypeDamageImmune(t))
+      || pokemon.getTag(BattlerTagType.UNDERGROUND, BattlerTagType.UNDERWATER)
+      || pokemon.switchOutStatus
+    ) {
+      return;
+    }
 
-    this.executeForAll((pokemon: Pokemon) => {
-      const immune: boolean =
-        !pokemon
-        || pokemon.getTypes(true, true).filter((t) => weather?.isTypeDamageImmune(t)).length > 0
-        || pokemon.switchOutStatus;
-      if (!immune) {
-        inflictDamage(pokemon);
-      }
-    });
+    const damage = toDmgValue(pokemon.getMaxHp() / 16);
```

Comment:
Every weather damage is always 1/16 of max-life? 
Can we put this value into a constant ?